File: app/src/main/java/fr/corenting/edcompanion/models/Ranks.java
package fr.corenting.edcompanion.models;

public class Ranks extends BaseModel {

    public Rank Combat;
    public Rank Trade;
    public Rank Explore;
    public Rank Cqc;

    public Rank Federation;
    public Rank Empire;

    public Ranks() {
        Combat = new Rank();
        Trade = new Rank();
        Explore = new Rank();
        Cqc = new Rank();
        Federation = new Rank();
        Empire = new Rank();
    }

    public class Rank {
        public String name;
        public int value;
        public int progress;
    }
}

File: app/src/main/java/fr/corenting/edcompanion/models/events/Ranks.kt
package fr.corenting.edcompanion.models.events

data class Ranks @JvmOverloads
constructor(val success: Boolean, val combat: Rank? = null, val trade: Rank? = null,
            val explore: Rank? = null, val cqc: Rank? = null, val federation: Rank? = null,
            val empire: Rank? = null) {

    data class Rank(val name: String, val value: Int, val progress: Int)
}
